src/bp/ui-studio/src/web/views/FlowBuilder/index.tsx

```diff
@@ -102,8 +102,10 @@ const FlowBuilder = (props: Props) => {
       f => f.currentMutex?.remainingSeconds && f.currentMutex.lastModifiedBy !== me
     )
 
-    setReadOnly(false)
-    setMutex(undefined)
+    if (isOperationAllowed({ operation: 'write', resource: 'bot.flows' })) {
+      setReadOnly(false)
+      setMutex(undefined)
+    }
```

Comment:
Somehow, I have to refresh the page at least one in order for the Flow Editor to be locked:


https://user-images.githubusercontent.com/9640576/117307106-48faa500-ae4e-11eb-8f0c-f04e22540a43.mp4